module/zfs/vdev.c

```diff
@@ -749,6 +749,11 @@ vdev_alloc(spa_t *spa, vdev_t **vdp, nvlist_t *nv, vdev_t *parent, uint_t id,
 		(void) nvlist_lookup_uint64(nv, ZPOOL_CONFIG_RESILVER_TXG,
 		    &vd->vdev_resilver_txg);
 
+		vd->vdev_resilver_deferred = (nvlist_lookup_boolean(nv,
+		    ZPOOL_CONFIG_RESILVER_DEFER) == 0);
```

Comment:
FYI, another way to do this would be `... = nvlist_exists(...);`. Not sure if that's any more or less clean.